### ISSUE TITLE
"Unexpected token at end of expression" in f-string

### ISSUE THREAD

**Issue by jgillard** [microsoft/pyright#1216]
## Environment data

-   Language Server version: 2020.11.2
-   OS and version: MacOS 10.15.7
-   Python version: 3.8.6

## Expected behaviour

When using string formatting alongside f-string debugging, no error should be shown as this code is valid in Python 3.8 and above.

Example: `f'{a = :.2f}'`

## Actual behaviour

A red underline appears inside this f-string.

<img width="198" alt="Screenshot 2020-12-01 at 12 09 05" src="https://user-images.githubusercontent.com/3770312/100738781-054acd80-33ce-11eb-8f41-7aa26ad43a11.png">

## Code Snippet

Input:
```python
a = 3.1415
print(f'{a:.2f}')
print(f'{a = }')
print(f'{a = :.2f}')
```

Output:
```
3.14
a = 3.1415
a = 3.14
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pyright & pylance.

**Comment by erictraut** [comment]
If you're interested in an interim workaround, if you delete the space between the "=" and the ":", the error will go away.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.91, which I just published. It will also be included in the next release of pylance.
